broker-cli/utils/basic-auth.spec.js

```diff
@@ -0,0 +1,78 @@
+const path = require('path')
+const { expect, rewire, sinon } = require('test/test-helper')
+
+const basicAuth = rewire(path.resolve(__dirname, 'basic-auth'))
+
+describe('basicAuth', () => {
+  describe('#credentialsToBasicAuth', () => {
+    let credentialsToBasicAuth
+    let user
+    let pass
+
+    beforeEach(() => {
+      user = 'sparkswap'
+      pass = 'password'
+      credentialsToBasicAuth = basicAuth.__get__('credentialsToBasicAuth')
+    })
+
+    it('generates a basic authentication header for a username and password', () => {
+      const res = credentialsToBasicAuth(user, pass)
+      // Format of the has is based off of RFC 7235
```

Comment:
Doesn't grpc have built-in username/pass? or do people recreate their own authentication headers like this?